tasks/install.yml

```diff
@@ -30,16 +30,17 @@
     state: directory
     mode: 0755
 
-- name: Create config dir
+- name: Create config and data directories
   file:
     path: "{{ item }}"
     state: directory
     owner: root
     group: "{{ promtail_system_group }}"
     mode: 0770
-  with_items:
+  loop:
     - "{{ promtail_config_dir }}"
     - "{{ promtail_config_file_sd_dir }}"
+    - "{{ promtail_positions_path | dirname }}"
```

Comment:
The proposed change would modify any existing directory - i.e. `promtail_positions_path: /tmp/positions.yml`.

I currently use `promtail_config_positions` to specify the filename directly :
```
promtail_config_positions:
  filename: /tmp/positions.yaml
```

This would not pick up the specified filename and still create a unnecessary directory.

Instead of creating a directory - I propose to rather create the file directly (`promtail_positions_path .filename`) with correct permissions for loki

Reply:
hm, i can't come up with something else than to create the directory conditionally when non-existing to cover both cases - using a common or a distinct directory.